core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultNettyOptions.java

```diff
@@ -74,6 +77,24 @@ public DefaultNettyOptions(InternalDriverContext context) {
             .setNameFormat(context.getSessionName() + "-admin-%d")
             .build();
     this.adminEventLoopGroup = new DefaultEventLoopGroup(adminGroupSize, adminThreadFactory);
+    // setup the Timer
+    ThreadFactory timerThreadFactory =
+        new ThreadFactoryBuilder().setThreadFactory(safeFactory).build();
+    timer =
+        new HashedWheelTimer(
+            timerThreadFactory,
+            config.getDuration(DefaultDriverOption.NETTY_TIMER_TICK_DURATION).toNanos(),
+            TimeUnit.NANOSECONDS,
+            config.getInt(DefaultDriverOption.NETTY_TIMER_TICKS_PER_WHEEL));
+    // In case things shutdown without calling close(), make sure the timer resources are released.
+    Runtime.getRuntime()
```

Comment:
I don't like having done this. However, some of the integration tests generate a bunch of warnings like this:
```
ERROR io.netty.util.ResourceLeakDetector - LEAK: HashedWheelTimer.release() was not called before it's garbage-collected. See http://netty.io/wiki/reference-counted-objects.html for more information.
```
I am calling `timer.stop` in the `onClose()` method below, but that does not seem to get called in certain ITs. After investigating the test and finding one that caused the warning, it seems that it happens when you attempt to create a new session with a misconfigured DriverConfigLoader. Doing so will:
1) create a DriverContext that has a lazy reference to an instance of NettyOptions, then
1) dereference the DriverContext's NettyOptions, which
1) calls the constructor for DefaultNettyOptions, which
1) creates the Timer instance (line above this one)

However since the test expects the session creation to fail, there is nothing left to invoke `onClose()`, so the Timer's resources are never released, as the warning indicates. The only way I was able to eliminate the warning was to add this shutdown hook.

Reply:
`DriverConfigValidationIT` was one that caused the warning. That test tries to create a session with a config loader that has an non-existent class specified for a variety of Driver options. If I comment out most of the calls to `should_fail_to_init_with_invalid_policy`, the warning doesn't appear. Seems like a race condition of sorts.

I didn't see any warnings about the event loops, but maybe they aren't tracked the same? I'll try to investigate what's going on some more.

Reply:
This looks like a bug in `DefaultSession`: `NettyOptions.onClose` should be invoked if init fails. Not clear yet why the unclosed event loops don't prevent the VM from exiting, I'm looking into it.